File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        
        return False
        

        # you can have a hashtable to store 
        # binary search tree 
        # time O(n) means within the bounds of the number of total elements
        # Space O(n) within the bounds of # of total elements

        # hashtable because the most we iterate is n. max space is the amount of n 

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        return bs(0, nums.length -1, nums, target);
    }

    public int bs(int l, int r, int[] nums, int target){
        if(l > r) return -1;
        int m = l + (r - l) / 2;

        if (nums[m] == target) return m;
        return (nums[m] < target) ? bs(m+1, r, nums, target): bs (l, m-1, nums, target);

    }
}
